build/mixxx.py

```diff
@@ -213,15 +213,15 @@ def __init__(self, target, machine, build, toolchain, available_features):
         # Now that environment variables have been read, we can detect the compiler.
         import subprocess
         process = subprocess.Popen("%s %s" %(self.env['CC'], '--version'), stdout=subprocess.PIPE, shell=True) # nosec
```

Comment:
`shell=True` in combination with user input opens the door for command injections, e.g. if `CC` is set to `rm -rf /; gcc`.
Instead, something like this should be used:

```python
process = subprocess.Popen([self.env['CC'], '--version'], stdout=subprocess.PIPE)`

// or:
stdout = subprocess.check_output([self.env['CC'], '--version'])`
```

Reply:
The shell=True argument has been added just recently by the NixOS build support, not sure why.

Reply:
@poelzi has explained it here: 
https://github.com/mixxxdj/mixxx/pull/2144/files#r291847013
Unfortunately the comment was not added to the code. Maybe we can add a comment here?

Reply:
In that case IMHO it's still undesirable to invoke a shell because we don't know what crazy non-standard `$SHELL` the user is using. `shlex.split()` can be used to split the content of CC env var, so that `shell=True` is not needed.

```python
stdout = subprocess.check_output([*shlex.split(self.env['CC']), '--version'])`

# or the longer version
cmd = shlex.split(self.env['CC'])
cmd.append('--version')
stdout = subprocess.check_output(cmd)`
 ```